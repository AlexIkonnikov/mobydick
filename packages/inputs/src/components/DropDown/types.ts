import {ReactElement} from 'react';

interface IDropDownProps {
  title?: string;
  rightIcon?: ReactElement;
  placeholder: string;
  list: string[];
  onPress: (item: string) => void;
  selectedItem?: string;
  dropDownWidth?: number;
  dropDownHeight?: number;
  navBarHeight?: number;
  // need for ignore ref in test
  jestTest?: boolean;
}

interface IDropDownIconProps {
  isOpen: boolean;
  rightIcon?: ReactElement | undefined;
}

export type DropDownProps = IDropDownProps;
export type DropDownIconProps = IDropDownIconProps;
