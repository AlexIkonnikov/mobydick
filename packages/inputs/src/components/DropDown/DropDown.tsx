import {
  View,
  TouchableHighlight,
  TouchableOpacity,
  ITouchableOpacity,
  FlatList,
} from '@npm/mobydick-core';
import React, {FC, useCallback, useRef, useState} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {Typography} from '@npm/mobydick-typography';
import {PopupBase, usePopups} from '@npm/mobydick-popups';
import {useDimensions} from '@react-native-community/hooks';
import {getModel} from 'react-native-device-info';

import {IDropDownProps} from './types';
import stylesCreate from './stylesCreate';
import {
  DEFAULT_DROP_DOWN_HEIGHT,
  DEFAULT_DROP_DOWN_WIDTH,
  DROP_DOWN_POPUP_ID,
} from './constants/constants';
import Icon from './components/DropDownIcon';
import getIosSafeAreaHeights from './constants/getIosSafeAreaHeights';
import {
  getDropDownDimensions,
  getDropDownHeights,
} from './constants/getDropDownDimensions';

const DropDown: FC<IDropDownProps> = props => {
  const {
    label,
    placeholder,
    list,
    selectedItem,
    onPress,
    rightIcon,
    addButtonStyle,
    addFlatListStyle,
    addFlatListItemStyle,
    navBarHeight = 50,
    addLabelStyle,
    addLabelFont,
    addButtonTextStyle,
    addButtonTextFont,
    addFlatListTextStyle,
    addFlatListTextFont,
    addFlatListTextFontPressed,
    addFlatListTextStylePressed,
    selectedItemColor,
    addButtonTextStyleChosen,
    addButtonTextFontChosen,
    maxVisibleListLength = 6,
  } = props;
  const {height} = useDimensions().window;
  const [styles, theme] = useStyles(stylesCreate);
  const [chosen, setChosen] = useState(selectedItem || '');
  const [pressedItem, setPressedItem] = useState('');
  const dropDownRef = useRef<ITouchableOpacity>(null);
  const [isOpen, setOpen] = useState(false);
  const popupContext = usePopups();
  const model = getModel();
  const {topIosMargin, bottomIosMargin} = getIosSafeAreaHeights(model);
  const {dropDownMaxHeight, dropDownItemHeight} = getDropDownHeights({
    dropDownHeight: addButtonStyle?.height
      ? +addButtonStyle.height
      : DEFAULT_DROP_DOWN_HEIGHT,
    flatListPaddingVertical: addFlatListStyle?.paddingVertical
      ? +addFlatListStyle.paddingVertical
      : styles.flatList.paddingVertical,
    listLength: list.length,
    maxVisibleListLength,
  });

  const checkPosition = () => {
    if (dropDownRef.current) {
      dropDownRef.current.measure((_x, _y, _width, _height, _pageX, pageY) => {
        openPopup(pageY);
        setOpen(true);
      });
    }
  };

  const keyExtractor = useCallback(
    (item: string, index: number) => index.toString() + item.toString(),
    [],
  );

  const openPopup = (pageY: number) => {
    const {listAbovePosition, listUnderPosition, expectedEndPositionOnScreen} =
      getDropDownDimensions({
        pageY,
        topIosMargin,
        navBarHeight,
        bottomIosMargin,
        maxVisibleListLength,
        dropDownHeight: addButtonStyle?.height
          ? +addButtonStyle.height
          : DEFAULT_DROP_DOWN_HEIGHT,
        flatListPaddingVertical: addFlatListStyle?.paddingVertical
          ? +addFlatListStyle.paddingVertical
          : styles.flatList.paddingVertical,
        listLength: list.length,
      });
    popupContext.openPopup({
      id: DROP_DOWN_POPUP_ID,
      Content: props => {
        return (
          <PopupBase
            onClose={() => {
              setOpen(false);
              props.onClose();
            }}
            overlayStyle={{backgroundColor: 'transparent'}}>
            <FlatList
              bounces={false}
              style={[
                styles.flatList,
                addFlatListStyle,
                {
                  width: addFlatListStyle?.width
                    ? addFlatListStyle.width
                    : addButtonStyle?.width
                    ? addButtonStyle.width
                    : DEFAULT_DROP_DOWN_WIDTH,
                },
                expectedEndPositionOnScreen > height
                  ? {top: listAbovePosition}
                  : {top: listUnderPosition},
                {
                  maxHeight: dropDownMaxHeight,
                },
              ]}
              data={list}
              renderItem={renderItem}
              keyExtractor={keyExtractor}
            />
          </PopupBase>
        );
      },
    });
  };

  const renderItemOnPress = (item: string) => {
    onPress(item);
    setPressedItem(item);
    setOpen(false);
    setChosen(item);
    popupContext.closePopup(DROP_DOWN_POPUP_ID);
  };

  const renderItem = ({item}: {item: string}) => {
    return (
      <TouchableHighlight
        style={[
          styles.dropDownItem,
          addFlatListItemStyle,
          {
            height: addFlatListItemStyle?.height
              ? addFlatListItemStyle.height
              : dropDownItemHeight,
          },
          item === pressedItem
            ? selectedItemColor
              ? {backgroundColor: selectedItemColor}
              : {backgroundColor: theme.colors.BgAccentSoft}
            : null,
        ]}
        onPress={() => renderItemOnPress(item)}
        underlayColor={
          selectedItemColor ? selectedItemColor : theme.colors.BgAccentSoft
        }>
        <Typography
          style={
            item === pressedItem
              ? addFlatListTextStylePressed
              : addFlatListTextStyle
          }
          font={
            item === pressedItem
              ? addFlatListTextFontPressed
                ? addFlatListTextFontPressed
                : 'Medium-Primary-M'
              : addFlatListTextFont
              ? addFlatListTextFont
              : 'Regular-Secondary-M'
          }>
          {item}
        </Typography>
      </TouchableHighlight>
    );
  };

  return (
    <View style={styles.container}>
      {Boolean(label) && (
        <Typography
          font={addLabelFont ? addLabelFont : 'Medium-Tertiary-XS'}
          style={addLabelStyle}>
          {label}
        </Typography>
      )}
      <View
        collapsable={false}
        ref={dropDownRef}
        style={{
          height: addButtonStyle?.height
            ? addButtonStyle.height
            : DEFAULT_DROP_DOWN_HEIGHT,
          width: addButtonStyle?.width
            ? addButtonStyle.width
            : DEFAULT_DROP_DOWN_WIDTH,
        }}>
        <TouchableOpacity
          style={[
            styles.button,
            addButtonStyle,
            {
              width: addButtonStyle?.width
                ? addButtonStyle.width
                : DEFAULT_DROP_DOWN_WIDTH,
            },
            {
              height: addButtonStyle?.height
                ? addButtonStyle.height
                : DEFAULT_DROP_DOWN_HEIGHT,
            },
            isOpen
              ? {
                  borderColor: addButtonStyle?.borderColor
                    ? addButtonStyle.borderColor
                    : theme.colors.BorderNormal,
                }
              : {
                  borderColor: addButtonStyle?.backgroundColor
                    ? addButtonStyle.backgroundColor
                    : theme.colors.BgSecondary,
                },
          ]}
          onPress={checkPosition}
          testID={'drop_down_button'}>
          <Typography
            style={chosen ? addButtonTextStyleChosen : addButtonTextStyle}
            font={
              chosen
                ? addButtonTextFontChosen
                  ? addButtonTextFontChosen
                  : 'Regular-Primary-M'
                : addButtonTextFont
                ? addButtonTextFont
                : 'Regular-Muted-M'
            }
            numberOfLines={1}>
            {chosen ? chosen : placeholder}
          </Typography>
          <Icon isOpen={isOpen} rightIcon={rightIcon} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default DropDown;
