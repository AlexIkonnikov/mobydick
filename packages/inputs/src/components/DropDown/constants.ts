import {Dimensions} from 'react-native';

export const {width, height} = Dimensions.get('window');
export const borderButtonWidth = 1;
export const maxVisibleFlatListItems = 6;
