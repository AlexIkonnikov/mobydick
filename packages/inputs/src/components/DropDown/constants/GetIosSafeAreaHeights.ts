import {Platform} from 'react-native';

import {IPhoneModelEnum, IMarginsForIos} from '../types';

import {
  BOTTOMBAR_DEFAULT_HEIGHT,
  BOTTOMBAR_IP_X_AND_AFTER_HEIGHT,
  STATUSBAR_DEFAULT_HEIGHT,
  STATUSBAR_IP12_13_HEIGHT,
  STATUSBAR_IP12M_13_MINI_HEIGHT,
  STATUSBAR_X_XS_11PRO_HEIGHT,
} from './constants';

const getIosSafeAreaHeights = (model: string): IMarginsForIos => {
  if (
    Platform.OS === 'android' ||
    (Platform.OS === 'ios' && (Platform.isPad || Platform.isTVOS))
  ) {
    return {topIosMargin: 0, bottomIosMargin: 0};
  }
  switch (model) {
    case IPhoneModelEnum.iPhoneX:
    case IPhoneModelEnum.iPhoneXS:
    case IPhoneModelEnum.iPhone11Pro:
    case IPhoneModelEnum.iPhoneXSMAX:
    case IPhoneModelEnum.iPhoneXR:
    case IPhoneModelEnum.iPhone11:
    case IPhoneModelEnum.iPhone11ProMax:
      return {
        topIosMargin: STATUSBAR_X_XS_11PRO_HEIGHT,
        bottomIosMargin: BOTTOMBAR_IP_X_AND_AFTER_HEIGHT,
      };
    case IPhoneModelEnum.iPhone12Pro:
    case IPhoneModelEnum.iPhone12:
    case IPhoneModelEnum.iPhone13:
    case IPhoneModelEnum.iPhone13Pro:
    case IPhoneModelEnum.iPhone12ProMax:
    case IPhoneModelEnum.iPhone13ProMax:
    case IPhoneModelEnum.iPhone14:
    case IPhoneModelEnum.iPhone14Pro:
    case IPhoneModelEnum.iPhone14ProMax:
    case IPhoneModelEnum.iPhone15:
    case IPhoneModelEnum.iPhone15Pro:
    case IPhoneModelEnum.iPhone15ProMax:
      return {
        topIosMargin: STATUSBAR_IP12_13_HEIGHT,
        bottomIosMargin: BOTTOMBAR_IP_X_AND_AFTER_HEIGHT,
      };
    case IPhoneModelEnum.iPhone12Mini:
    case IPhoneModelEnum.iPhone13Mini:
    case IPhoneModelEnum.iPhone14Mini:
    case IPhoneModelEnum.iPhone15Mini:
      return {
        topIosMargin: STATUSBAR_IP12M_13_MINI_HEIGHT,
        bottomIosMargin: BOTTOMBAR_IP_X_AND_AFTER_HEIGHT,
      };
    default:
      return {
        topIosMargin: STATUSBAR_DEFAULT_HEIGHT,
        bottomIosMargin: BOTTOMBAR_DEFAULT_HEIGHT,
      };
  }
};

export default getIosSafeAreaHeights;
