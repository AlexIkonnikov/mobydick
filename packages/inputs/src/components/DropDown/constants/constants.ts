export const borderButtonWidth = 1;
export const dropDownListItemMultiplier = 0.8;
export const dropDownPopupId = 'DropDownPopup';
export const STATUSBAR_DEFAULT_HEIGHT = 20;
export const STATUSBAR_X_XS_11PRO_HEIGHT = 44;
export const STATUSBAR_IP12_13_HEIGHT = 47;
export const STATUSBAR_IP12M_13_MINI_HEIGHT = 50;
export const BOTTOMBAR_DEFAULT_HEIGHT = 0;
export const BOTTOMBAR_IP_X_AND_AFTER_HEIGHT = 34;
export const defaultDropDownHeight = 48;
export const defaultDropDownWidth = 200;
