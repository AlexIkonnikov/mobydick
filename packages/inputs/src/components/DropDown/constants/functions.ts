import {Platform} from 'react-native';

import {IMarginsForIos} from '../types';

import {
  BOTTOMBAR_DEFAULT_HEIGHT,
  BOTTOMBAR_IP11_HEIGHT,
  IP12_HEIGHT,
  IP12_WIDTH,
  IP12MAX_HEIGHT,
  IP12MAX_WIDTH,
  STATUSBAR_DEFAULT_HEIGHT,
  STATUSBAR_IP12_HEIGHT,
  STATUSBAR_IP12MAX_HEIGHT,
  STATUSBAR_X_HEIGHT,
  X_HEIGHT,
  X_WIDTH,
  XSMAX_HEIGHT,
  XSMAX_WIDTH,
} from './constants';

const getIosSafeAreaHeights = (
  width: number,
  height: number,
): IMarginsForIos => {
  let statusBarHeight = STATUSBAR_DEFAULT_HEIGHT;
  let bottomBarHeight = BOTTOMBAR_DEFAULT_HEIGHT;
  if (
    Platform.OS === 'android' ||
    (Platform.OS === 'ios' && (Platform.isPad || Platform.isTVOS))
  ) {
    return {topIosMargin: 0, bottomIosMargin: 0};
  }
  if (width === X_WIDTH && height === X_HEIGHT) {
    statusBarHeight = STATUSBAR_X_HEIGHT;
    bottomBarHeight = BOTTOMBAR_IP11_HEIGHT;
  } else if (width === XSMAX_WIDTH && height === XSMAX_HEIGHT) {
    statusBarHeight = STATUSBAR_X_HEIGHT;
    bottomBarHeight = BOTTOMBAR_IP11_HEIGHT;
  } else if (width === IP12MAX_WIDTH && height === IP12MAX_HEIGHT) {
    statusBarHeight = STATUSBAR_IP12MAX_HEIGHT;
    bottomBarHeight = BOTTOMBAR_IP11_HEIGHT;
  } else if (width === IP12_WIDTH && height === IP12_HEIGHT) {
    statusBarHeight = STATUSBAR_IP12_HEIGHT;
    bottomBarHeight = BOTTOMBAR_IP11_HEIGHT;
  }

  return {topIosMargin: statusBarHeight, bottomIosMargin: bottomBarHeight};
};

export default getIosSafeAreaHeights;
