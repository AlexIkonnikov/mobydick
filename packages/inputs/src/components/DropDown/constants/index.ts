export const BORDER_BUTTON_WIDTH = 1;
export const DROP_DOWN_LIST_ITEM_MULTIPLIER = 0.8;

export const DROP_DOWN_POPUP_ID = 'DropDownPopup';

export const STATUSBAR_DEFAULT_HEIGHT = 20;
export const STATUSBAR_X_XS_11PRO_HEIGHT = 44;
export const STATUSBAR_IP12_13_HEIGHT = 47;
export const STATUSBAR_IP12M_13_MINI_HEIGHT = 50;

export const BOTTOMBAR_DEFAULT_HEIGHT = 0;
export const BOTTOMBAR_IP_X_AND_AFTER_HEIGHT = 34;

export const DEFAULT_DROP_DOWN_HEIGHT = 48;
export const DEFAULT_DROP_DOWN_WIDTH = 200;
export const DEFAULT_DROPDOWN_MARGIN_FROM_BUTTON = 8;

export const ACCESSIBILITY_LABEL = {
  selector: 'selector',
};
