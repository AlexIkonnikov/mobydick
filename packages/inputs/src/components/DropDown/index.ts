import DropDown from './DropDown';
export * from './types';
export {DropDown};
