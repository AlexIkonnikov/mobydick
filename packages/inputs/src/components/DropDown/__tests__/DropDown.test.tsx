import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {SimpleIcon} from '@npm/mobydick-styles';
import {PopupsProvider} from '@npm/mobydick-popups';
import {View} from 'react-native';
import Constants from '@npm/mobydick-popups/src/components/PopupBase/constants';

import DropDown from '../DropDown';
import {ACCESSIBILITY_LABEL} from '../constants';
import {ITypes} from '../../types';

describe('@npm/mobydick-inputs/DropDown', () => {
  let viewRef: React.RefObject<View>;
  beforeEach(() => {
    viewRef = React.createRef();
    render(<View ref={viewRef} />);
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  const list = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10'];

  it('renders correctly 10 elements', () => {
    const {toJSON} = render(
      <PopupsProvider>
        <DropDown
          selectedItem={list[1]}
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={list}
          onPress={jest.fn()}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('selectItem', async () => {
    const onPress = jest.fn();
    const array = [
      {label: 'Русский', value: 'Русский'},
      {label: 'English', value: 'English'},
    ] as const;
    jest
      .spyOn(viewRef.current as View, 'measure')
      .mockImplementation(
        (
          cb: (
            x: number,
            y: number,
            width: number,
            height: number,
            pageX: number,
            pageY: number,
          ) => void,
        ) => cb(0, 0, 1, 1, 287, 2410),
      );

    const {getByLabelText} = render(
      <PopupsProvider>
        <DropDown
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={[...array]}
          onPress={onPress}
          selectedItem={array[1]}
          type={ITypes.disabled}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    const pressable = getByLabelText(ACCESSIBILITY_LABEL.selector);
    fireEvent.press(pressable);

    const pressableSelect = getByLabelText(array[1].label);
    fireEvent.press(pressableSelect);

    expect(onPress).toHaveBeenCalledWith(array[1]);
  });

  it('selectItemClose', async () => {
    const onPress = jest.fn();
    const array = ['Русский', 'English'];
    const useRefSpy = jest
      .spyOn(viewRef.current as View, 'measure')
      .mockImplementation(
        (
          cb: (
            x: number,
            y: number,
            width: number,
            height: number,
            pageX: number,
            pageY: number,
          ) => void,
        ) => cb(0, 0, 1, 1, 287, 2410),
      );

    const {getByLabelText, toJSON, getByTestId} = render(
      <PopupsProvider>
        <DropDown
          selectedItem={'selectedItem'}
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={array}
          onPress={onPress}
          subtitle={'subtitle'}
          subtitleIcon={'icon-account'}
          buttonStyle={{
            width: 400,
            height: 70,
            borderColor: '#000',
            backgroundColor: '#000',
          }}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    const pressable = getByLabelText(ACCESSIBILITY_LABEL.selector);

    fireEvent.press(pressable);

    const pressableClose = getByTestId(Constants.testID);
    const copyTarget = 'copyTarget';
    fireEvent.press(pressableClose, {
      target: copyTarget,
      currentTarget: copyTarget,
    });
    expect(toJSON()).toMatchSnapshot();
    expect(useRefSpy).toHaveBeenCalledTimes(1);
  });

  it('renders correctly 10 elements type wrong', () => {
    const {toJSON} = render(
      <PopupsProvider>
        <DropDown
          selectedItem={'selectedItem'}
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={list}
          onPress={jest.fn()}
          type={ITypes.wrong}
          labelFont={'Medium-Tertiary-XS'}
          buttonStyle={{
            width: 400,
            height: 70,
            borderColor: '#000',
            backgroundColor: '#000',
          }}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly 10 elements type valid', () => {
    const {toJSON} = render(
      <PopupsProvider>
        <DropDown
          selectedItem={'selectedItem'}
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={list}
          onPress={jest.fn()}
          type={ITypes.valid}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly 10 elements disabled', () => {
    const {toJSON} = render(
      <PopupsProvider>
        <DropDown
          selectedItem={undefined}
          placeholder={'Выберите язык'}
          label={'Название поля'}
          list={list}
          onPress={jest.fn()}
          disabled={true}
          rightIcon={<SimpleIcon name={'icon-arrow-down'} />}
        />
      </PopupsProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
