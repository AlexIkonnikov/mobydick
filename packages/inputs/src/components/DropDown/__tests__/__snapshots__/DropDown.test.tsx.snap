// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-inputs/DropDown renders correctly 10 elements 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        Array [
          Object {
            "paddingBottom": 12,
          },
          undefined,
        ],
      ]
    }
  >
    Название поля
  </Text>
  <View
    collapsable={false}
  >
    <View
      accessibilityLabel="selector"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EEEFF1",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 72,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "opacity": 1,
          "paddingHorizontal": 18,
          "width": 300,
        }
      }
    >
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        2
      </Text>
      <View
        style={false}
      >
        <Text
          allowFontScaling={true}
          style={
            Object {
              "color": "#5E6678",
              "fontFamily": "Neotis",
              "fontSize": 36,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/DropDown renders correctly 10 elements disabled 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        Array [
          Object {
            "paddingBottom": 12,
          },
          undefined,
        ],
      ]
    }
  >
    Название поля
  </Text>
  <View
    collapsable={false}
  >
    <View
      accessibilityLabel="selector"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D1D4DB",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 72,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "opacity": 1,
          "paddingHorizontal": 18,
          "width": 300,
        }
      }
    >
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#9BA1B0",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        Выберите язык
      </Text>
      <View
        style={false}
      >
        <Text
          allowFontScaling={true}
          style={
            Object {
              "color": "#5E6678",
              "fontFamily": "Neotis",
              "fontSize": 36,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/DropDown renders correctly 10 elements type valid 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        Array [
          Object {
            "paddingBottom": 12,
          },
          undefined,
        ],
      ]
    }
  >
    Название поля
  </Text>
  <View
    collapsable={false}
  >
    <View
      accessibilityLabel="selector"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(43, 120, 238, 0.15)",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 72,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "opacity": 1,
          "paddingHorizontal": 18,
          "width": 300,
        }
      }
    >
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        selectedItem
      </Text>
      <View
        style={false}
      >
        <Text
          allowFontScaling={true}
          style={
            Object {
              "color": "#5E6678",
              "fontFamily": "Neotis",
              "fontSize": 36,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/DropDown renders correctly 10 elements type wrong 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        Array [
          Object {
            "paddingBottom": 12,
          },
          undefined,
        ],
      ]
    }
  >
    Название поля
  </Text>
  <View
    collapsable={false}
  >
    <View
      accessibilityLabel="selector"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderColor": "#000",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "opacity": 1,
          "paddingHorizontal": 18,
          "width": 400,
        }
      }
    >
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        selectedItem
      </Text>
      <View
        style={false}
      >
        <Text
          allowFontScaling={true}
          style={
            Object {
              "color": "#5E6678",
              "fontFamily": "Neotis",
              "fontSize": 36,
            }
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-inputs/DropDown selectItemClose 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#5E6678",
          "fontFamily": "Inter-Medium",
          "fontSize": 21,
          "lineHeight": 27,
        },
        Array [
          Object {
            "paddingBottom": 12,
          },
          undefined,
        ],
      ]
    }
  >
    Название поля
  </Text>
  <View
    collapsable={false}
  >
    <View
      accessibilityLabel="selector"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "borderColor": "#000",
          "borderRadius": 12,
          "borderWidth": 1.5,
          "flexDirection": "row",
          "height": 70,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "opacity": 1,
          "paddingHorizontal": 18,
          "width": 400,
        }
      }
    >
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#20242D",
              "fontFamily": "Inter-Regular",
              "fontSize": 24,
              "lineHeight": 30,
            },
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ],
          ]
        }
      >
        selectedItem
      </Text>
      <View
        style={false}
      >
        <Text
          allowFontScaling={true}
          style={
            Object {
              "color": "#5E6678",
              "fontFamily": "Neotis",
              "fontSize": 36,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          Object {
            "color": "#9BA1B0",
            "fontFamily": "Neotis",
            "fontSize": 24,
            "marginRight": 6,
          }
        }
      >
        
      </Text>
      <Text
        allowFontScaling={true}
        style={
          Array [
            Object {
              "color": "#9BA1B0",
              "fontFamily": "Inter-Regular",
              "fontSize": 18,
              "lineHeight": 24,
            },
            undefined,
          ]
        }
      >
        subtitle
      </Text>
    </View>
  </View>
</View>
`;
